techlibs/gatemate/synth_gatemate.cc

```diff
@@ -0,0 +1,393 @@
+ *  WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ *  MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ *  ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ *  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ *  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ *  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include "kernel/register.h"
+#include "kernel/celltypes.h"
+#include "kernel/rtlil.h"
+#include "kernel/log.h"
+
+USING_YOSYS_NAMESPACE
+PRIVATE_NAMESPACE_BEGIN
+
+struct SynthGateMatePass : public ScriptPass
+{
+	SynthGateMatePass() : ScriptPass("synth_gatemate", "synthesis for Cologne Chip GateMate FPGAs") { }
+
+	void help() override
+	{
+		//   |---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|
+		log("\n");
+		log("    synth_gatemate [options]\n");
+		log("\n");
+		log("This command runs synthesis for Cologne Chip AG GateMate FPGAs.\n");
+		log("\n");
+		log("    -top <module>\n");
+		log("        use the specified module as top module.\n");
+		log("\n");
+		log("    -vlog <file>\n");
+		log("        write the design to the specified verilog file. Writing of an output\n");
+		log("        file is omitted if this parameter is not specified.\n");
+		log("\n");
+		log("    -blif <file>\n");
+		log("        write the design to the specified BLIF file. Writing of an output file\n");
+		log("        is omitted if this parameter is not specified.\n");
+		log("\n");
+		log("    -edif <file>\n");
+		log("        write the design to the specified EDIF file. Writing of an output file\n");
+		log("        is omitted if this parameter is not specified.\n");
+		log("\n");
+		log("    -json <file>\n");
+		log("        write the design to the specified JSON file. Writing of an output file\n");
+		log("        is omitted if this parameter is not specified.\n");
+		log("\n");
+		log("    -run <from_label>:<to_label>\n");
+		log("        only run the commands between the labels (see below). An empty\n");
+		log("        from label is synonymous to 'begin', and empty to label is\n");
+		log("        synonymous to the end of the command list.\n");
+		log("\n");
+		log("    -noflatten\n");
+		log("        do not flatten design before synthesis.\n");
+		log("\n");
+		log("    -nobram\n");
+		log("        do not use CC_BRAM_20K or CC_BRAM_40K cells in output netlist.\n");
+		log("\n");
+		log("    -noaddf\n");
+		log("        do not use CC_ADDF full adder cells in output netlist.\n");
+		log("\n");
+		log("    -nomult\n");
+		log("        do not use CC_MULT multiplier cells in output netlist.\n");
+		log("\n");
+		log("    -nomx8, -nomx4\n");
+		log("        do not use CC_MX{8,4} multiplexer cells in output netlist.\n");
+		log("\n");;
+		log("    -dff\n");
+		log("        run 'abc'/'abc9' with -dff option\n");

```

Comment:
Since you don't run `abc9` ever, might as well delete it from this help message to avoid confusion.